lib/plugins/package/lib/packageService.js

```diff
@@ -120,19 +120,21 @@ module.exports = {
 
     // use the artifact in function config if provided
     if (funcPackageConfig.artifact) {
-      const filePath = path.join(this.serverless.config.servicePath, funcPackageConfig.artifact);
+      const filePath = path.resolve(this.serverless.config.servicePath, funcPackageConfig.artifact);
       functionObject.package.artifact = filePath;
       return filePath;
     }
 
     // use the artifact in service config if provided
     // and if the function is not set to be packaged individually
     if (this.serverless.service.package.artifact && !funcPackageConfig.individually) {
-      const filePath = path.join(
+      const filePath = path.resolve(
         this.serverless.config.servicePath,
         this.serverless.service.package.artifact
       );
       funcPackageConfig.artifact = filePath;
+      functionObject.package = funcPackageConfig;
```

Comment:
This change is a bit controversial as it's changed only because I needed a way to perform an assertion in tests for `service-wide` scenario - do you think there's another way of doing it without introducing extra production code?

Reply:
Why exactly do we need it?

Technically in tests we should purely test the outcome, so either inspect generated artifact, or generated CF template

Reply:
Would you suggest to assert on `Properties.Code.S3Key` on CF function resources instead here?

Reply:
> Would you suggest to assert on Properties.Code.S3Key on CF function resources instead here?

As I think more of it, that will also not work, as we just blindly assign there basename of artifact. So fact that full path is broken won't be exposed.

If I understand correctly we're fixing here issue that happens during deployment, where for upload to S3 we try to reach wrong path.

In light of that probably best test is one that tests against `sls deploy` and `sls deploy function` commands and confirms that we attempt to upload expected content to S3.

I think we can construct it by mocking AWS SDK through [awsRequestStubMap](https://github.com/serverless/test/blob/master/docs/run-serverless.md#awsrequeststubmap`) option, and by setting [lastLifecycleHookName](https://github.com/serverless/test/blob/master/docs/run-serverless.md#lastlifecyclehookname) to avoid full deployment process.
e.g. we have test that does something similar here: https://github.com/serverless/serverless/blob/ef530506d5044ab3312c829838bb29cfcd2c889f/test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js#L1010

As hint you may inspect what exactly lifecycle events are triggered in which order by running deployment with [`debug`](https://github.com/serverless/serverless/tree/debug) branch (I've just ensured it's up to date with master). It logs all meta about that (we will put into core with regular debug logs after we have #1720 addressed)

Reply:
For `sls deploy function` we're just passing artifact data to `updateFunctionCode`, so I've went this route for these tests (ensuring that AWS API is called with correct content). As for `sls deploy`, I believe it's not affected by this issue, but I'll try to come up with meaningful tests for that scenario as well

Reply:
I've adjusted the approach a bit as confirming on the content was a bit cumbersome in case of readStream, so I've decided to confirm that the stream is created based on proper path to artifact